src/functions/users/flextime/handler.ts

```diff
@@ -32,34 +22,57 @@ export const listUsersFlextimeHandler: APIGatewayProxyHandler = async (event) =>
     const usersWithFlextime = await Promise.allSettled(
       optedInUsers.map(async (severaUser) => {
         try {
+          /**
+           * TODO: This should be typed according to the spec response type for the lambda function
+           * Spec in the lambdas for the futur
+           */
           const flextime = await api.getFlextimeBySeveraUserId(severaUser.guid);
           return {
             user: {
               id: severaUser.guid,
               firstName: severaUser.firstName || "",
               lastName: severaUser.lastName || "",
-              email: severaUser.email || "",
-              attributes: {
-                severaUserId: severaUser.guid,
-                isActive: true
-              }
             },
             flextime: {
               totalFlextimeBalance: flextime?.totalFlextimeBalance || 0,
               monthFlextimeBalance: flextime?.monthFlextimeBalance || 0
             }
           };
         } catch (error) {
-          throw error;
+          /**
+           * Per-user error handling
+           */
+          console.error(`Error fetching flextime for user ${severaUser.guid}:`, error);
+          return {
+            user: {
+              id: severaUser.guid,
+              firstName: severaUser.firstName || "",
+              lastName: severaUser.lastName || "",
+              attributes: {
+                severaUserId: severaUser.guid,
+                isActive: true
+              }
+            },
+            flextime: null,
+            error: true,
+            /**
+             * Find the appropriate status code for the error
+             */
+            statusCode: error instanceof Error && (error as any).statusCode ? (error as any).statusCode : 500,
+            message: `Failed to fetch flextime for user ${severaUser.guid}: ${error instanceof Error ? error.message : "Unknown error"}`
```

Comment:
Mostly right but this should be done in the same structure as the catch return example at the bottom of this file- line 79